### DISCUSSION TITLE
Static source for grid square geometries?

### DISCUSSION THREAD

**Discussion by jkeefe** [weather-gov/api#621] [Q&A] [answered]
Is the source material that provides the geometry for every ID/X/Y square available somewhere — as a vector, raster, GRIB2, netcdf or other static file? We'd like to map every square with their IDs, but can't find that reference anywhere.

Thank you!

**Comment by sullynole** [comment, marked as answer]
The gridpoints WFO/x/y should not be considered static as the "grid id" can change at any time when a WFO adjusts their grid domain.  The points endpoint exists to always map to the current gridpoint.  If looking for a gridded forecast, suggest looking at NDFD: https://vlab.noaa.gov/web/mdl/ndfd

**Reply by randallspicher** [reply]
Does that mean we should re-fetch the points url (https://api.weather.gov/points/[lat],[long]) every time because the x/y in the forecast urls can change at any time?  Currently, in my app I fetch the points during the setup (asking the user to enter their lat/long) , and then store the forecast urls (and station/zone info) in the settings, so-as to only need to need to make one http call later on to get the actual forecast.  How often do those grid coords change?  (if I added some logic to recheck once a day or something like that)

**Reply by sullynole** [reply]
Typically a WFO grid domain could go months or even years without changing, but it is possible a WFO could change their domain one day so depending on your application it may be best to recheck regularly (such as daily).  Most of the grid alignments were just adjusted with the last API release so there could be application changes that require a recheck as well.
